go-client/client/query.go

```diff
@@ -1054,13 +1445,30 @@ func (op mergeOp) makeBatchOp(resultId *ticketpb2.Ticket, children []*tablepb2.T
 	return tablepb2.BatchTableRequest_Operation{Op: &tablepb2.BatchTableRequest_Operation_Merge{Merge: req}}
 }
 
-// Merge combines two or more tables into one aggregate table.
+func (op mergeOp) execSerialOp(ctx context.Context, stub *tableStub, children []*TableHandle) (*TableHandle, error) {
+	assert(len(children) == len(op.children), "wrong number of children for Merge")
+
+	return stub.merge(ctx, op.sortBy, children)
+}
+
+func (op mergeOp) String() string {
+	return fmt.Sprintf("Merge(%s, /* %d tables omitted */)", op.sortBy, len(op.children))
+}
+
+// MergeQuery combines two or more tables into one table as part of a query.
 // This essentially appends the tables one on top of the other.
 // If sortBy is provided, the resulting table will be sorted based on that column.
-func (qb QueryNode) Merge(sortBy string, others ...QueryNode) QueryNode {
-	children := make([]QueryNode, len(others)+1)
-	children[0] = qb
-	copy(children[1:], others)
+func MergeQuery(sortBy string, tables ...QueryNode) QueryNode {
+	if len(tables) == 0 {
+		// A merge with no tables will always error,
+		// but since we can't return an error from this method
+		// we have to let the server give us an error instead.
+		emptyNode := QueryNode{builder: &queryBuilder{}}
+		return emptyNode.addOp(mergeOp{children: tables, sortBy: sortBy})
+	}
+
+	// Just pick an arbitrary node to add the operation to.
+	qb := tables[0]
```

Comment:
In the "normal" `merge` code, it is possible and valid to pass in `null` for tables.  If that is possible here, the base node needs to be searched for, or null tables need to be removed from tables.